src/main/scala/com/fulcrumgenomics/commons/reflect/ReflectionUtil.scala

```diff
@@ -396,13 +396,15 @@ object ReflectionUtil {
         val ctor: Constructor[_] = clazz.getDeclaredConstructor(classOf[String])
         ctor.setAccessible(true)
         ctor.newInstance(value)
-      }.getOrElse {
+      }.orElse( Try {
         // Last chance: if everything else fails look to see if there is a companion object
         // with an apply(String) method that could work
         val typ             = mirror.classSymbol(clazz).toType
         val companionMirror = mirror.reflectModule(typ.typeSymbol.companion.asModule)
         val companion       = companionMirror.instance
         companion.getClass.getMethod("apply", classOf[String]).invoke(companion, value)
+      }).getOrElse {
+        throw new ReflectionException(s"No suitable single-argument constructor found to construct a '${unitType.getSimpleName}' from a string: $value")
```

Comment:
This isn't necessarily true - the constructor could have existed, but the call failed.